src/jit/lsra.cpp

```diff
@@ -2643,8 +2643,7 @@ bool LinearScan::isMatchingConstant(RegRecord* physRegRecord, RefPosition* refPo
         switch (otherTreeNode->OperGet())
         {
             case GT_CNS_INT:
-                if ((refPosition->treeNode->AsIntCon()->IconValue() == otherTreeNode->AsIntCon()->IconValue()) &&
-                    (varTypeGCtype(refPosition->treeNode) == varTypeGCtype(otherTreeNode)))
```

Comment:
The comment for the change that added the second check in TFS was:

> (emitThisGCrefRegs & regMask) == 0' assert failure.
> 
> The problem is that we were reusing a constant zero that was a TYP_REF.
> The fix is to not reuse a register if it is not the same GC type.

Reply:
Ah, an assert, that doesn't necessarily mean that this change won't work. I also changed codegen to never emit a GC constant 0 so that assert should never be hit. Though I may need to do the same thing for non-0 byrefs, I'll check.

Thanks for the info!

Reply:
Yes, when @sandreenko mentioned this to me, I suspected that the assert may have been overly aggressive. Given the changes to codegen, it would seem that this is a safe change.